content/articles/bigdecimal-and-primitive-data-type-arithmetic-program-in-java/index.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /bigdecimal-and-primitive-data-type-arithmetic-program-in-java/
+title: BigDecimal and Primitive Data Type Arithmetic Program in Java
+description: This articles introduces the reader to the basic arithmetic operations that can be performed on BigDecimal as well as between BigDecimal and primitive data types.
+author: bonface-ndolo
+date: 2021-09-22T00:00:00-03:40
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/bigdecimal-and-primitive-data-type-arithmetic-program-in-java/hero.jpg
+    alt: BigDecimal and Primitive Data Type Arithmetic Program in Java Hero Image
+---
+
+Because of their inaccuracy, floating-point data types cannot be used in financial calculations. This is why Java provides a separate class named **BigDecimal** for executing operations and reducing the chances of calculation errors. 

```

Comment:
```suggestion
Because of their inaccuracy, floating-point data types cannot be used in financial calculations. This is why Java provides a separate class named BigDecimal for executing operations. And, reducing the chances of calculation errors.
```